lib/instabug_http_client.dart

```diff
@@ -10,11 +10,10 @@ import 'package:instabug_http_client/instabug_http_logger.dart';
 import 'package:meta/meta.dart';
 
 class InstabugHttpClient extends InstabugHttpLogger implements http.Client {
-  InstabugHttpClient() : client = http.Client() {
+  InstabugHttpClient({http.Client? client}) : client = client ?? http.Client() {
     logger = this;
   }
```

Comment:
The fact that both the parameter and the attribute are called `client` is a bit confusing, so I'd say let's change the class attribute to `_client` to avoid confusion.

Reply:
@a7medev , I think I disagree with this recommendation. From an API perspective this is a completely expected Dart pattern. For example, if we were to write this constructor in a very common Dart way:
```dart
InstabugHttpClient({this.client}) {
  _logger - this.
}

final http.Client? client;
```
Consumers of this package would then be able to use the constructor in either of the following two ways:
```dart
// zero arg
InstabugHttpClient();
```
OR
```dart
// with custom inner client
InstabugHttpClient(client: myClient);
```

The way I've written the constructor in this PR allows for exactly the same usage.

Furthermore, changing the class variable to `_client` would be a breaking API change and absolutely would require a major version upgrade in pubspec.yaml. Given the previous code's @visibleForTesting annotation I think it would break tests. I recommend against this suggested change.

If we would like to distinguish between the InstabugHttpClient and the inner client provided by the caller, we could change the field to `innerClient` or something similar. However, this would still require a major version bump according to SemVer (non-backward compatible breaking change).

Please re-review. I think the API usage provided will feel natural to users and this implementation will not require a major version bump.

Reply:
@dustin-graham I think that's reasonable, let's continue with it this way then.